Enable RT1170

Lucien-Zhao opened the pull request:
RT1170 enablement on zephyr has been prepared.

jhqian committed:
dts: arm: nxp: update rt1170 dts

- update soc dtsi

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
soc: arm: nxp_imx: add rt117x support

1. Added RT10xx and RT11xx configs
2. Added a new soc file for rt117x. There are clock differences
   between the RT10xx and RT11xx series, hence the soc files
   have been separated.

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
drivers: clock_control: add ccm driver for rt117x

- add ccm driver for rt117x

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
board: arm: add board support files for mimxrt1170_evk

Add board support files for mimxrt1170_evk

- Add pinmux, dts and doc
- Tested samples: hello_world, philosophers, synchronization,
  basic/blinky, basic/button.

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

lucien-nxp committed:
west: update the head of hal_nxp

- to get the update hal_nxp from PR #76

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

galak reviewed (changes requested):
Add a dts/arm/nxp/nxp_rt1176*.dtsi set of files that the board dtsi files include.  This way when other variants of the 117x are supported we don't have to change things.

galak commented on drivers/clock_control/CMakeLists.txt:
```diff
@@ -39,3 +38,11 @@ else()
 endif()
 endif()
+
+if(CONFIG_CLOCK_CONTROL_MCUX_CCM)
```
maybe better to just introduce a 117x specific Kconfig symbol.

galak reviewed (changes requested):
Feels like the "nxp,imx-ccm" for 117x should not use the same binding/compatible and introduce its own "nxp,imx-rt117x-ccm" or something like that.

galak commented on soc/arm/nxp_imx/rt117x/soc.c:
```diff
@@ -0,0 +1,393 @@
+	/* disable interrupts */
+	oldLevel = irq_lock();
+
+#if __CORTEX_M == 7
```
Should use `#ifdef CONFIG_SOC_MIMXRT1176_CM7` here

mmahadevan108 replied:
Done.

galak commented on soc/arm/nxp_imx/rt117x/Kconfig.soc:
```diff
@@ -0,0 +1,196 @@
+	  that you should not set directly. The part number selection choice defines
+	  the default value for this string.
+
+config CM7_FREQ
```
Why isn't this coming from devicetree?

galak reviewed (commented):
How much of what is in the the nxp_rt1170_cm{4,7}.dtsi is specified by hardware v user choice?  seems like a lot of that belongs in the board dtsi files and not the common "soc" versions.

Lucien-Zhao commented:
> Add a dts/arm/nxp/nxp_rt1176*.dtsi set of files that the board dtsi files include. This way when other variants of the 117x are supported we don't have to change things.

MaureenHelm commented:
@Lucien-Zhao why did you close this PR?

Lucien-Zhao commented:
> @Lucien-Zhao why did you close this PR?

Sorry, I don't be familiar with GITHUB operation. My original intention was to reply comment.

laodzu commented:
Cool stuff!  I am really looking forward to i.MXRT1170 support!

Trying hello_world for either CM4 or CM7 yields this device tree error:

> - Found BOARD.dts: /opt/src/git/zephyrproject/zephyr/boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm4.dts
devicetree error: 'gptfreq' is marked as required in 'properties:' in /opt/src/git/zephyrproject/zephyr/dts/bindings/timer/nxp,imx-gpt.yaml, but does not appear in <Node /soc/gpt@400ec000 in 'mimxrt1170_evk_cm4.dts.pre.tmp'>

And indeed gptfreq is not defined in the .dts file.

Also I wonder why `dts/arm/nxp/nxp_rt.dtsi` is not used here.  By the filename I was under the impression this is common to all i.MXRT parts.  If it is, then it should be usable also by the i.MXRT117x but if it isn't, then it should be renamed into a more specific filename.

MaureenHelm commented on dts/arm/nxp/nxp_rt1170_cm4.dtsi:
```diff
@@ -0,0 +1,54 @@
+
+/ {
+	cpus {
+		cpu@1 {
+			compatible = "arm,cortex-m4";
+		};
```
Is this needed?

MaureenHelm commented on soc/arm/nxp_imx/rt117x/Kconfig.series:
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2020, NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_IMX_RT117X
```
I'd really like to see if we can do this without introducing a new SoC series. There is a lot of code duplicated from `soc/arm/nxp_imx/rt/`, so I would prefer if we can add RT117x there.

MaureenHelm commented on boards/arm/mimxrt1170_evk/board.cmake:
```diff
@@ -0,0 +1,11 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+board_runner_args(pyocd "--target=mimxrt1170_cm7")
+board_runner_args(jlink "--device=MIMXRT1176xxxA_M7")
```
Will this work for the M4 too?

Lucien-Zhao replied:
I have modified as lpcxpresso55s69 doing.

MaureenHelm commented on boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm4.dts:
```diff
@@ -0,0 +1,54 @@
+	compatible = "nxp,mimxrt1176";
+
+	aliases {
+		led0 = &green_led;
+		sw0 = &user_button;
```
You can move the leds and buttons to a common dtsi, then include it in the cm4 and cm7 dts files.

See `boards/arm/lpcxpresso55s69/lpcxpresso55s69.dtsi` for example.

Lucien-Zhao replied:
Thank you for reminding me. I have modified.

MaureenHelm reviewed (changes requested):
Thanks @Lucien-Zhao !

> Also I wonder why dts/arm/nxp/nxp_rt.dtsi is not used here. By the filename I was under the impression this is common to all i.MXRT parts. If it is, then it should be usable also by the i.MXRT117x but if it isn't, then it should be renamed into a more specific filename.

I have the same question. I haven't compared them in detail (are the peripheral addresses the same?) but `nxp_rt.dtsi` and `nxp_rt117x.dtsi` look really similar. It looks like you could `nxp_rt.dtsi` and then just add the rt117x-specific bits.

Lucien-Zhao commented:
> Thanks @Lucien-Zhao !
> 
> > Also I wonder why dts/arm/nxp/nxp_rt.dtsi is not used here. By the filename I was under the impression this is common to all i.MXRT parts. If it is, then it should be usable also by the i.MXRT117x but if it isn't, then it should be renamed into a more specific filename.
> 
> I have the same question. I haven't compared them in detail (are the peripheral addresses the same?) but `nxp_rt.dtsi` and `nxp_rt117x.dtsi` look really similar. It looks like you could `nxp_rt.dtsi` and then just add the rt117x-specific bits.

RT1170 is a new product line and there are many changes on peripheral addresses. As I know from the moment, the double core is the biggest feature of RT11xx series. It is necessary to separate 1170 from the rt series in my perspective.

mmahadevan108 commented on boards/arm/mimxrt1170_evk/doc/index.rst:
```diff
@@ -0,0 +1,280 @@
+| ENET      | on-chip    | ethernet                            |
++-----------+------------+-------------------------------------+
+| USB       | on-chip    | USB device controller               |
++-----------+------------+-------------------------------------+
```
Can you confirm USB, ENET, SDHC, DISPLAY are enabled. I don't see these in the dts file

Lucien-Zhao replied:
I have checked index.rst file again and deleted peripherals disabled.

MaureenHelm commented:
> > > Also I wonder why dts/arm/nxp/nxp_rt.dtsi is not used here. By the filename I was under the impression this is common to all i.MXRT parts. If it is, then it should be usable also by the i.MXRT117x but if it isn't, then it should be renamed into a more specific filename.
> > 
> > 
> > I have the same question. I haven't compared them in detail (are the peripheral addresses the same?) but `nxp_rt.dtsi` and `nxp_rt117x.dtsi` look really similar. It looks like you could `nxp_rt.dtsi` and then just add the rt117x-specific bits.
> 
> RT1170 is a new product line and there are many changes on peripheral addresses. As I know from the moment, the double core is the biggest feature of RT11xx series. It is necessary to separate 1170 from the rt series in my perspective.

Even if we can't extract common parts of the dts, I'd still like to see if we can merge `soc/arm/nxp_imx/rt117x` into `soc/arm/nxp_imx/rt` to avoid all the duplicated Kconfig symbols.

mmahadevan108 commented on dts/arm/nxp/nxp_rt1170_cm4.dtsi:
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2020, NXP
```
Year should be 2021

mmahadevan108 commented on dts/arm/nxp/nxp_rt1170_cm7.dtsi:
```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020, NXP
+ *
```
Year should be 2021

mmahadevan108 commented on dts/arm/nxp/nxp_rt117x.dtsi:
```diff
@@ -0,0 +1,790 @@
+/*
+ * Copyright (c) 2020, NXP
+ *
```
2021. Please fix all instances

mmahadevan108 commented on drivers/clock_control/clock_control_mcux_ccm_rt117x.c:
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
```
Year

mmahadevan108 commented on drivers/clock_control/CMakeLists.txt:
```diff
@@ -3,6 +3,7 @@
 zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_BEETLE              beetle_clock_control.c)
 zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_LPC11U6X            clock_control_lpc11u6x.c)
 zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_MCUX_CCM            clock_control_mcux_ccm.c)
+zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_MCUX_CCM_RT117X     clock_control_mcux_ccm_rt117x.c)
```
Can we call this CCM_VER2 instead of tying this with RT117X.

MaureenHelm commented:
@Lucien-Zhao when do you expect to return to this PR? The merge window for the 2.6 release will close on May 7.

Lucien-Zhao commented:
> window



> @Lucien-Zhao when do you expect to return to this PR? The merge window for the 2.6 release will close on May 7.

I will resolve this issue recently. We was preparing rt1160 release before.

mmahadevan108 commented:
Hello @Lucien-Zhao any update

mmahadevan108 commented:
@Lucien-Zhao Any update?

CristXu commented:
Hi, 
    I want to check with you about both the sdram and the on board flash's size:
![image](https://user-images.githubusercontent.com/49855018/120928313-bed88180-c716-11eb-91bf-fb4adc4b0ed3.png)
![image](https://user-images.githubusercontent.com/49855018/120928335-d7489c00-c716-11eb-9f8d-396fcb435257.png)
  Maybe here is somethin wrong? 
  I double checked that the sdram is 64MB and the qspi flash is 16MB on the rt1170_evk, and also the qspi flash is IS25WP128 instead, not IS25WP064.

Regards,
    Crist

Lucien-Zhao commented:
> > > > Also I wonder why dts/arm/nxp/nxp_rt.dtsi is not used here. By the filename I was under the impression this is common to all i.MXRT parts. If it is, then it should be usable also by the i.MXRT117x but if it isn't, then it should be renamed into a more specific filename.
> > > 
> > > 
> > > I have the same question. I haven't compared them in detail (are the peripheral addresses the same?) but `nxp_rt.dtsi` and `nxp_rt117x.dtsi` look really similar. It looks like you could `nxp_rt.dtsi` and then just add the rt117x-specific bits.
> > 
> > 
> > RT1170 is a new product line and there are many changes on peripheral addresses. As I know from the moment, the double core is the biggest feature of RT11xx series. It is necessary to separate 1170 from the rt series in my perspective.
> 
> Even if we can't extract common parts of the dts, I'd still like to see if we can merge `soc/arm/nxp_imx/rt117x` into `soc/arm/nxp_imx/rt` to avoid all the duplicated Kconfig symbols.

Should I remove all rt117x series and add rt1170 device into rt series?  Or move rt117x series into rt series to reduce the Kconfig symbols?

Lucien-Zhao commented:
> Hi,
> I want to check with you about both the sdram and the on board flash's size:
> ![image](https://user-images.githubusercontent.com/49855018/120928313-bed88180-c716-11eb-91bf-fb4adc4b0ed3.png)
> ![image](https://user-images.githubusercontent.com/49855018/120928335-d7489c00-c716-11eb-9f8d-396fcb435257.png)
> Maybe here is somethin wrong?
> I double checked that the sdram is 64MB and the qspi flash is 16MB on the rt1170_evk, and also the qspi flash is IS25WP128 instead, not IS25WP064.
> 
> Regards,
> Crist

Thank you, you are right. I have modified them in dtsi files.

mmahadevan108 commented:
I have add the RT1170 changes to the `soc/arm/nxp_imx/rt` folder and deleted `soc/arm/nxp_imx/rt11xx` folder. Also I have renamed the new CCM clock driver.

galak commented on dts/arm/nxp/nxp_rt1170_cm4.dtsi:
```diff
@@ -0,0 +1,50 @@
+				<12 0>, <13 0>, <14 0>, <15 0>,
+				<16 0>;
+		};
+
```
remove blank lines.

galak commented on soc/arm/nxp_imx/rt/Kconfig.soc:
```diff
@@ -9,6 +9,9 @@ choice
 
 config SOC_MIMXRT1011
 	bool "SOC_MIMXRT1011"
+	select CPU_CORTEX_M7
```
should we put CPU_CORTEX_M7 and CPU_CORTEX_M_HAS_DWT under SOC_SERIES_IMX_RT10XX instead of duplicating them?

mmahadevan108 replied:
Thanks, I will make this change.

galak reviewed (changes requested):
various cleanups requested.

galak reviewed (changes requested):
@mmahadevan108 we should name the devicetree for CCM something like:

`nxp,imx-rt117x-ccm` and have the driver be something like clock_control_mcux_rt117x_ccm.c

mmahadevan108 commented:
> @mmahadevan108 we should name the devicetree for CCM something like:
> 
> `nxp,imx-rt117x-ccm` and have the driver be something like clock_control_mcux_rt117x_ccm.c

Sure, I will change to nxp,imx-ccm-rev2 as this version of the CCM module is not limited to RT117X family. It will be used in future SoC's such as RT1160 etc.

galak commented:
> Sure, I will change to nxp,imx-ccm-rev2 as this version of the CCM module is not limited to RT117X family. It will be used in future SoC's such as RT1160 etc.

If its really called `rev2` from your guys point of view, than I'm fine with that.

mmahadevan108 commented on boards/arm/mimxrt1170_evk/mimxrt1170_evk.dtsi:
```diff
@@ -0,0 +1,41 @@
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button: button-1 {
+		  label = "User SW7";
```
Done

mmahadevan108 replied:
Done

mmahadevan108 commented on boards/arm/mimxrt1170_evk/CMakeLists.txt:
```diff
@@ -0,0 +1,14 @@
+
+if (CONFIG_DISPLAY)
+message(WARNING "
+CONFIG_DISPLAY: Running this firmware on a board without a display may damage the board
```
Removed

mmahadevan108 committed:
drivers: hwinfo: Update i.MX RT Hwinfo driver

Add support for RT1170. The fuse register that holds
the unique device ID is different as compared to the
RT10XX series

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 commented on boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.yaml:
```diff
@@ -0,0 +1,18 @@
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 262144
```
Thanks, I updated the values for CM4 and CM7

mmahadevan108 commented on soc/arm/nxp_imx/rt117x/soc.c:
```diff
@@ -0,0 +1,393 @@
+	}
+#endif
+
+#if __CORTEX_M == 4
```
Done

MaureenHelm commented on boards/arm/mimxrt1170_evk/doc/index.rst:
```diff
@@ -0,0 +1,237 @@
+   https://www.nxp.com.cn/docs/en/data-sheet/IMXRT1170CEC.pdf
+
+.. _i.MX RT1170 Reference Manual:
+   https://www.nxp.com/docs/en/reference-manual/IMXRT1170RM.pdf
```
this link doesn't work

mmahadevan108 replied:
Thank you, I have fixed it.

MaureenHelm commented on boards/arm/mimxrt1170_evk/evkmimxrt1170_connect_cm4_cm4side.jlinkscript:
```diff
@@ -0,0 +1,93 @@
+ * Copyright 2019-2020 NXP
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
```
Not sure why the scancode check isn't failing on this, @nashif?

I'd like to see if we can relicense these files to Apache, but in the meantime let's remove them from this PR and add back later in a follow up PR.

mmahadevan108 replied:
Removed this file

MaureenHelm reviewed (changes requested):
Can you add more detail to the commit messages?

mmahadevan108 commented on boards/arm/mimxrt1170_evk/Kconfig.defconfig:
```diff
@@ -0,0 +1,24 @@
+	default "mimxrt1170_evk_cm7" if BOARD_MIMXRT1170_EVK_CM7
+	default "mimxrt1170_evk_cm4" if BOARD_MIMXRT1170_EVK_CM4
+
+if BOARD_MIMXRT1170_EVK_CM7
+choice CODE_LOCATION
+	default CODE_FLEXSPI
+endchoice
+endif #BOARD_MIMXRT1170_EVK_CM7
+
+if BOARD_MIMXRT1170_EVK_CM4
+choice CODE_LOCATION
+	default CODE_SRAM0
+endchoice
+endif #BOARD_MIMXRT1170_EVK_CM4
```
Thank you. I have made the change.

mmahadevan108 commented on soc/arm/nxp_imx/rt/linker.ld:
```diff
@@ -11,8 +11,8 @@
      {
-#if (DT_REG_SIZE(DT_NODELABEL(ocram)) > 0) && !IS_CHOSEN_SRAM(ocram)
-        OCRAM  (wx) : ORIGIN = DT_REG_ADDR(DT_NODELABEL(ocram)), LENGTH = DT_REG_SIZE(DT_NODELABEL(ocram))
+#if (DT_REG_SIZE(DT_NODELABEL(ocram0)) > 0) && !IS_CHOSEN_SRAM(ocram0)
+        OCRAM  (wx) : ORIGIN = DT_REG_ADDR(DT_NODELABEL(ocram0)), LENGTH = DT_REG_SIZE(DT_NODELABEL(ocram0))
```
I have removed this change.

mmahadevan108 commented on dts/arm/nxp/nxp_rt117x.dtsi:
```diff
@@ -0,0 +1,790 @@
+			reg = <0x40c34000 0x4000>;
+			interrupts = <36 0>;
+			clocks = <&ccm IMX_CCM_LPI2C5_CLK 0x80 24>;
+			label = "I2C_4";
```
Thank you for spotting this.

mmahadevan108 commented on soc/arm/nxp_imx/rt/Kconfig.soc:
```diff
@@ -374,14 +480,40 @@ config HAS_ARM_DIV
+config CM7_FREQ
+	int "CM7 frequency (integer) in MHz"
+	range 624 996
+	default 996
```
Removed this config and added to the dts file.

mmahadevan108 commented on boards/arm/mimxrt1170_evk/doc/index.rst:
```diff
@@ -0,0 +1,237 @@
+   https://www.nxp.com/design/development-boards/i-mx-evaluation-and-development-boards/i-mx-rt1170-evaluation-kit:MIMXRT1170-EVK
+
+.. _MIMXRT1170-EVK Board Hardware User's Guide:
+   https://www.nxp.com/docs/en/user-guide/MIMXRT1170EVKHUG.pdf
```
Thank you, I have fixed it.

mmahadevan108 commented on boards/arm/mimxrt1170_evk/doc/index.rst:
```diff
@@ -0,0 +1,237 @@
+Overview
+********
+
+The i.MX RT1170 crossover MCUs are setting speed records at 1 GHz.
+This ground-breaking family combines superior computing power and multiple
```
I have updated this paragraph

mmahadevan108 commented:
> How much of what is in the the nxp_rt1170_cm{4,7}.dtsi is specified by hardware v user choice? seems like a lot of that belongs in the board dtsi files and not the common "soc" versions.

@galak I have reviewed the nxp_rt1170_cm{4,7}.dtsi files. These choices were made based on how the memories and IP blocks are connected in hardware. DTCM and ITCM are available only to the CM7 code, SRAM's are available only to the CM4 core. EDMA interrupts are connected to specific cores and hence have been moved to specific files. 

GPIO6 and 9 interrupts are connected to specific cores. I have made a few updates to the dtsi files.

mmahadevan108 commented on boards/arm/mimxrt1170_evk/doc/index.rst:
```diff
@@ -0,0 +1,233 @@
+   https://www.nxp.com/products/processors-and-microcontrollers/arm-microcontrollers/i-mx-rt-crossover-mcus/i-mx-rt1170-crossover-mcu-family-first-ghz-mcu-with-arm-cortex-m7-and-cortex-m4-cores:i.MX-RT1170
+
+.. _i.MX RT1170 Datasheet:
+   https://www.nxp.com.cn/docs/en/data-sheet/IMXRT1170CEC.pdf
```
Done, thank you for spotting this.

mmahadevan108 commented:
@galak @MaureenHelm. Could you please take a look. I believe all review comments have been addressed.

mmahadevan108 committed:
dt-bindings: clock: Update for i.MX CCM Rev 2

- update clock macro for CCM
- Add a dts bindings file for CCM Rev 2

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

MaureenHelm commented on soc/arm/nxp_imx/rt/CMakeLists.txt:
```diff
@@ -1,12 +1,10 @@
-  soc.c
-  )
+zephyr_sources_ifdef(CONFIG_SOC_SERIES_IMX_RT11XX soc_rt11xx.c)
+zephyr_sources_ifdef(CONFIG_SOC_SERIES_IMX_RT10XX soc_rt10xx.c)
```
Please explain in the commit message why you're splitting up soc.c (i.e., significantly different clocking)

mmahadevan108 replied:
Done

galak commented:
> > 
> 
> @galak I have reviewed the nxp_rt1170_cm{4,7}.dtsi files. These choices were made based on how the memories and IP blocks are connected in hardware. DTCM and ITCM are available only to the CM7 code, SRAM's are available only to the CM4 core. EDMA interrupts are connected to specific cores and hence have been moved to specific files.
> 
> GPIO6 and 9 interrupts are connected to specific cores. I have made a few updates to the dtsi files.

Maybe good to add some top level comment in the .dtsi files to says this and maybe makes any references to the sections in the manual to that document it

mmahadevan108 commented:
I have update the west.yml to point to the latest NXP HAL

MaureenHelm commented:
@galak please revisit

mmahadevan108 commented:
Anything else needed for this PR?

galak commented on dts/arm/nxp/nxp_rt1170_cm4.dtsi:
```diff
@@ -0,0 +1,67 @@
+
+&gpio6 {
+	/* Dummy IRQ since no IRQ is routed to CM4 and this field is marked as required */
+	interrupts = <199 0>;
```
we should change the binding to not require `interrupts` instead of adding a dummy.

galak commented on dts/arm/nxp/nxp_rt1170_cm7.dtsi:
```diff
@@ -0,0 +1,73 @@
+
+&gpio9 {
+	/* Dummy IRQ since no IRQ is routed to CM7 and this field is marked as required */
+	interrupts = <199 0>;
```
same comment about adding dummys

galak commented on dts/arm/nxp/nxp_rt1170_cm7.dtsi:
```diff
@@ -0,0 +1,73 @@
+
+&gpio11 {
+	/* Dummy IRQ since no IRQ is routed to CM7 and this field is marked as required */
+	interrupts = <200 0>;
```
same comment about adding dummys

mmahadevan108 replied:
I have updated the bindings to not require interrupts. I had to make a small change to the GPIO driver file as well as it was looking for interrupts.

galak commented on boards/arm/mimxrt1170_evk/board.cmake:
```diff
@@ -0,0 +1,21 @@
+
+
+if(CONFIG_SOC_MIMXRT1176_CM4)
+board_runner_args(pyocd "--target=mimxrt1170_cm4")
```
refs to jlinkscript dont make sense since that got dropped.

mmahadevan108 replied:
Thank you for pointing this. I have removed this section.

galak reviewed (changes requested):
in general LGTM, a few minor things need fixing.

mmahadevan108 committed:
dts: bindings: imx-gpio: Do not require interrupts

Some SoC's do not have interrupts connected to the GPIO module

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
drivers: igpio: Do not require GPIO interrupts to be present

Some SoC's do not have a GPIO interrupt connected

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

galak commented on dts/arm/nxp/nxp_rt1170_cm4.dtsi:
```diff
@@ -0,0 +1,62 @@
+};
+
+&gpio9 {
+	interrupts = <99 0>;
```
why is this 99?

Lucien-Zhao replied:
Because there is no fixed interrupt number assigned to GPIO9. 
![image](https://user-images.githubusercontent.com/68273245/123799291-a7a64180-d91a-11eb-938f-6243a157795c.png)

galak replied:
ah, thanks

galak commented on dts/arm/nxp/nxp_rt1170_cm4.dtsi:
```diff
@@ -0,0 +1,62 @@
+
+&gpio11 {
+	/* GPIO 7-11 share the same interrupt on the CM4 */
+	interrupts = <99 0>;
```
why is this 99?

galak reviewed (commented):
LGTM.  One question about use of IRQ 99 in nxp_rt1170_cm4.dtsi.
